includes/User.php

```diff
@@ -4841,7 +4841,29 @@ protected function saveOptions() {
 			}
 		}
 
-		$dbw->delete( 'user_properties', array( 'up_user' => $this->getId() ), __METHOD__ );
+		// kinda ghetto, but :(
+		if ($wgPreferencesUseService) {
+			$regex = implode ('|', $wgUserPreferenceWhiteList['regexes']);
```

Comment:
Well I agree on the concept but getPreferences() would return only preferences set for the user (at a given time) not the current list of preferences managed by user-preferences service. I would go into making getter for this in userPreferences so it will return the list of known preferences... Still there is a case where we should handle regex'es.